src/geometry/euclidean/sphere.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import geometry.euclidean.basic
+
+/-!
+# Spheres
+
+This file proves basic geometrical results about distances and angles
+in spheres in real inner product spaces and Euclidean affine spaces.
+-/
+
+open real
+open_locale euclidean_geometry real_inner_product_space real
+
+variables {V : Type*} [inner_product_space ℝ V]
+
+namespace inner_product_geometry
+
+/-!
+### Geometrical results on spheres in real inner product spaces
+
+This section develops some results on spheres in real inner product spaces,
+which are used to deduce corresponding results for Euclidean affine spaces.
+-/
+
+lemma mul_norm_eq_abs_sub_sq_norm_of_smul_and_norm_eq_norm {x y z : V}
+  (h₁ : ∃ k : ℝ, k ≠ 1 ∧ x + y = k • (x - y)) (h₂ : ∥z - y∥ = ∥z + y∥) :
+  ∥x - y∥ * ∥x + y∥ = abs (∥z + y∥ ^ 2 - ∥z - x∥ ^ 2) :=
+begin
+  obtain ⟨k, hk_ne_one, hk⟩ := h₁,
+  let r := (k - 1)⁻¹ * (k + 1),
+
+  have hxy : x = r • y,
+  { rw [← smul_smul, eq_inv_smul_iff' (sub_ne_zero.mpr hk_ne_one), ← sub_eq_zero],
+    calc  (k - 1) • x - (k + 1) • y
+        = (k • x - x) - (k • y + y) : by simp_rw [sub_smul, add_smul, one_smul]
+    ... = (k • x - k • y) - (x + y) : by simp_rw [← sub_sub, sub_right_comm]
+    ... = k • (x - y) - (x + y)     : by rw ← smul_sub k x y
+    ... = 0                         : sub_eq_zero.mpr hk.symm },
+
+  have hzy : ⟪z, y⟫ = 0,
+  { rw [← eq_of_pow_two_eq_pow_two (norm_nonneg (z - y)) (norm_nonneg (z + y)),
+        norm_add_pow_two_real, norm_sub_pow_two_real] at h₂,
+    linarith },
+
+  have hzx : ⟪z, x⟫ = 0 := by rw [hxy, inner_smul_right, hzy, mul_zero],
+
+  symmetry,
```

Comment:
Is there a reason you want this `symmetry`? Instead of just reversing the `calc`?

Reply:
Done!